internal/statesync/reactor.go

```diff
@@ -126,6 +134,8 @@ type Reactor struct {
 	// received snapshots and chunks into the sync.
 	mtx    tmsync.RWMutex
 	syncer *syncer
+
+	stateProvider StateProvider // provides state data for bootstrapping the StateSyncReactor
```

Comment:
What was the reason for moving the stateProvider to be part of the reactor struct and not just an argument that gets passed when a node calls `Sync()`? I guess constructing it internally makes it more difficult for users to construct their own stateProvider but it seems like we were already heading in that direction

Reply:
@cmwaters didn't think in this way, I will move the state provider init before calling `startStateSync` and pass it into `Sync()`
Thanks for the suggestion!

Reply:
@cmwaters 
ah, the reason I cannot pass StateProvider to `Sync()` was the mock test. There will be the import cycle in `MockSyncReactor`

Reply:
ok, I solved the mock test issue.